core/src/test/java/io/substrait/plan/PlanConverterTest.java

```diff
@@ -0,0 +1,106 @@
+import io.substrait.utils.StringHolderHandlingProtoAdvancedExtensionConverter;
+import org.junit.jupiter.api.Test;
+
+class PlanConverterTest {
+  @Test
+  void emptyAdvancedExtensionTest() {
+    final Plan plan = Plan.builder().advancedExtension(AdvancedExtension.builder().build()).build();
+    final PlanProtoConverter toProtoConverter = new PlanProtoConverter();
+    final io.substrait.proto.Plan protoPlan = toProtoConverter.toProto(plan);
+
+    final ProtoPlanConverter fromProtoConverter = new ProtoPlanConverter();
+    final Plan plan2 = fromProtoConverter.from(protoPlan);
+
+    assertEquals(plan, plan2);
+  }
+
+  @Test
+  void enhancementOnlyAdvancedExtension() {
+    final StringHolder enhanced = new StringHolder("ENHANCED");
+
+    final Plan plan =
+        Plan.builder()
+            .advancedExtension(AdvancedExtension.builder().enhancement(enhanced).build())
+            .build();
+    final PlanProtoConverter toProtoConverter = new PlanProtoConverter();
+    final io.substrait.proto.Plan protoPlan = toProtoConverter.toProto(plan);
+
+    final ProtoPlanConverter fromProtoConverter = new ProtoPlanConverter();
+    assertThrows(IllegalStateException.class, () -> fromProtoConverter.from(protoPlan));
+  }
+
+  @Test
+  void optimizationOnlyAdvancedExtension() {
+    final StringHolder optimized = new StringHolder("OPTIMIZED");
+
+    final Plan plan =
+        Plan.builder()
+            .advancedExtension(AdvancedExtension.builder().addOptimizations(optimized).build())
+            .build();
+    final PlanProtoConverter toProtoConverter = new PlanProtoConverter();
+    final io.substrait.proto.Plan protoPlan = toProtoConverter.toProto(plan);
+
+    final ProtoPlanConverter fromProtoConverter = new ProtoPlanConverter();
+    final Plan plan2 = fromProtoConverter.from(protoPlan);
+
+    // The optimization is serialized correctly to protobuf.
+    // When it is read back in, the default ProtoPlanConverter drops it.
+    // As such they are not equal anymore.
+    assertNotEquals(plan, plan2);
+  }
+
+  @Test
+  void advancedExtensionWithEnhancementAndOptimization() {
+    final StringHolder enhanced = new StringHolder("ENHANCED");
+    final StringHolder optimized = new StringHolder("OPTIMIZED");
+
+    final Plan plan =
+        Plan.builder()
+            .advancedExtension(
+                AdvancedExtension.builder()
+                    .enhancement(enhanced)
+                    .addOptimizations(optimized)
+                    .build())
+            .build();
+    final PlanProtoConverter toProtoConverter = new PlanProtoConverter();
+    final io.substrait.proto.Plan protoPlan = toProtoConverter.toProto(plan);
+
+    final ProtoPlanConverter fromProtoConverter = new ProtoPlanConverter();
+
+    // Enhancements are not handled by the default ProtoPlanConverter
+    assertThrows(IllegalStateException.class, () -> fromProtoConverter.from(protoPlan));
+  }
+
+  @Test
+  void customAdvancedExtensionSerDes() {
```

Comment:
```suggestion
  void customAdvancedExtensionSerde() {
```

This is me being incredibly opinionated on a thing that doesn't really matter, but I saw `SerDes` and it made my brain stop. It feels wrong.

Reply:
I have no problem renaming this one